scripts/tf_cnn_benchmarks/benchmark_cnn.py

```diff
@@ -3002,12 +3003,17 @@ def setup(params):
       formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
   # Sets environment variables for MKL
+  # Will set to defaults only if either cmd line arguments are missing or
+  # OS ENV variables are missing. 
   if params.mkl:
-    os.environ['KMP_BLOCKTIME'] = str(params.kmp_blocktime)
-    os.environ['KMP_SETTINGS'] = str(params.kmp_settings)
-    os.environ['KMP_AFFINITY'] = params.kmp_affinity
-    if params.num_intra_threads > 0:
-      os.environ['OMP_NUM_THREADS'] = str(params.num_intra_threads)
+    if '-kmp_blocktime' in str(sys.argv) or 'KMP_BLOCKTIME' not in os.environ:
```

Comment:
I like the 0 default as that is a much better default, but I would rather not set these from the envar because I want the value set to what  am passing for tests.  If for some reason the environment has these values set I would not want them to mess up my unit tests where I am testing X value.  The driving reason is this code is used for testing and as a sandbox.

Reply:
I realize it is only taking the ENV if I do not set it, but I expect it to use my default if I am not setting it.

Reply:
Please see the example below, I have written comments in the code below.
I hope this clears any confusion. Let me know, if this breaks any of your unit tests.

export KMP_SETTINGS=1
python tf_cnn_benchmarks.py  --batch_size=1 --model=googlenet \
--device=cpu --mkl=True --kmp_blocktime=10  

  ```if params.mkl:
    if '-kmp_blocktime' in str(sys.argv) or 'KMP_BLOCKTIME' not in os.environ:
        os.environ['KMP_BLOCKTIME'] = str(params.kmp_blocktime)  # This line gets executes, kmp_blocktime is set to 10
    if '-kmp_settings' in str(sys.argv) or 'KMP_SETTINGS' not in os.environ:
        os.environ['KMP_SETTINGS'] = str(params.kmp_settings) # This line does not get execute, takes from OS ENV
    if '-kmp_affinity' in str(sys.argv) or 'KMP_AFFINITY' not in os.environ:
        os.environ['KMP_AFFINITY'] = params.kmp_affinity # This line gets executed and sets the default value
    if params.num_intra_threads > 0 or 'OMP_NUM_THREADS' not in os.environ:
        os.environ['OMP_NUM_THREADS'] = str(params.num_intra_threads) # This line gets executed and sets the default value```

Reply:
I understand what you are saying.  I do not want ENV_VARS to overwrite the defaults, it leads to situations where it is not clear what value was used for not a lot of added value.  In your example KMP_SETTINGS was set although the default is 0.  Pretty minor in this case.

Reply:
Could we then, print a warning to let the user know that OS ENV_VARS was ignored and script defaults are being used.
This might clear confusion from both ends :)

Reply:
A warning is fine. Also, if you want to use the env var by default, you can do, e.g.:

```
export KMP_SETTINGS=1
python tf_cnn_benchmarks.py --kmp_settings=${KMP_SETTINGS:-0}
```